src/lib/core/TLVReader.cpp

```diff
@@ -330,6 +331,25 @@ CHIP_ERROR TLVReader::Get(CharSpan & v) const
     }
 
     v = CharSpan(Uint8::to_const_char(bytes), len);
+#if CHIP_CONFIG_TLV_VALIDATE_CHAR_STRING_ON_READ
```

Comment:
I would rather do that separately.
Ideally we would have these enabled always, however I believe realistically controllers cannot (because backwards compatibility)  and by extension neither can devices that need to talk to older devices.

Runtime means adding setters and getters and storage and maybe tests. I do not want to keep increasing the scope of the PR.

Reply:
OK, but we absolutely need to make this runtime-possible as a follow-up...

Reply:
Right now, this new code path is not directly unit tested in TLV unit tests because it's not runtime and there are no getters to conditionalize the tests.

Reply:
I was relying that every integration test gets this because chip-tool now has it enabled by default, so we should have significant coverage here.

Reply:
created #30399 for this.